presto-main/src/main/java/com/facebook/presto/operator/aggregation/BindableAggregationFunction.java

```diff
@@ -164,16 +151,25 @@ public InternalAggregationFunction specialize(BoundVariables variables, int arit
                 factory);
     }
 
-    private AccumulatorStateSerializer<?> getAccumulatorStateSerializer(BoundVariables variables, TypeManager typeManager, FunctionRegistry functionRegistry, AggregationImplementation implementation, Class<?> stateClass, Optional<Method> stateSerializerFactory, DynamicClassLoader classLoader)
+    // TODO Extract for scalar use case
+    private static MethodHandle bindDependencies(Method function, List<ImplementationDependency> dependencies, BoundVariables variables, TypeManager typeManager, FunctionRegistry functionRegistry)
+            throws IllegalAccessException
+    {
+        MethodHandle inputHandle = lookup().unreflect(function);
```

Comment:
`input` is misleading here. The function is called for input/combine/output. (And you're going to extract it for scalar use.)